src/traffic-editor.md

```diff
@@ -1,3 +1,522 @@
+the path to this file from the dialog box that opens from from clicking `Edit ->
+Project Properties`. The project can be assigned a name from the same box.
+Saving (`Ctrl + S`) the project will write the configuration to the
+`tutorial.project.yaml` file while populating the `tutorial.building.yaml` file
+with a basic template. Additional `.building.yaml` files may be created for
+other building in the site. The project configuration can be updated to
+reference the new file accordingly.
+
+### Adding a level
+A new level in the building can be added by clicking the `Add` button in the
+`levels` tab of the `Sidebar`. The operation will open a dialog box where the
+`name`, `elevation` (in meters) and path to a 2D `drawing` file (`.png`) can be
+specified. In most use cases, the floor plan for the level is used as the
+drawing. If unspecified, the user may explicitly enter dimensions of the level
+in the fields provided.
+
+![](images/traffic_editor/add_level.png)
+
+In the figure above, a new level `L1` at `0m` elevation and a floor plan has
+been added as reflected in the `levels` tab. A default scale fo `1px = 5cm` is
+applied. The actual scale can be set by adding a measurement. Any offsets
+applied to align levels will be reflected in the `X` and `Y` columns. Saving the
+project will update the `tutorial.building.yaml` files as seen below.
+```yaml
+
+levels:
+  L1:
+    drawing:
+      filename: office.png
+    elevation: 0
+    flattened_x_offset: 0
+    flattened_y_offset: 0
+    layers:
+      {}
+lifts:
+  {}
+name: building
+```
+### Adding a vertex
+|                    Icon                    | Shortkey |
+|:------------------------------------------:| :-------:| 
+| ![](images/traffic_editor/icons/vertex.svg)| `v` |
+
+A vertex is a fundamental component of multiple annotations. Walls,
+measurements, doors, floor polygons and traffic lanes are created from two or
+more vertices. To create a vertex, click on the vertex icon in the `Main
+Toolbar` and anywhere on the canvas. The default attributes of a vertex are its
+coordinates along with an empty name field. Additional attributes may be added
+by clicking the `Add` button in the figure. Short descriptions of these are
+presented below.
+* **is_partking_spot:** if true and if the waypoint is part of a traffic lane,
+  the `rmf_fleet_adapter` will treat this as a _holding point_ during path
+  planning, ie, the robot is allowed to wait at this waypoint for an indefinite
+  period of time.
+* **is_charger:** if true and if the waypoint is part of a traffic lane, the
+  `rmf_fleet_adapter` will treat this as a charing station.
+* **dock_name:** if specified and if the waypoint is part of a traffic lane, the
+  `rmf_fleet_adapter` will issue an `rmf_fleet_msgs::ModeRequest` message with
+  `MODE_DOCKING` and `task_id` equal to the specified name to the robot as it
+  approaches this waypoint.
+* **workcell_name [deprecated]:** This attribute was previously used by the
```

Comment:
If this is already deprecated and this is our first release of the documentation, should we even include this?

Reply:
That is a good point. I added this note since the gui still has this option. Perhaps we can remove it when a future release of traffic editor gets rid of this option?

Reply:
sounds like a plan! :+1: